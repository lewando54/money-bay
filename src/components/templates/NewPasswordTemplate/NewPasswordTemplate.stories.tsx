/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import NewPasswordTemplate from './NewPasswordTemplate'

const meta = {
	component: NewPasswordTemplate,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof NewPasswordTemplate>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
