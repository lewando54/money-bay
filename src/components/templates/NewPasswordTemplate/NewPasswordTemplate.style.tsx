import { Dimensions, StyleSheet } from 'react-native'

import globalThemeStyle, { MAIN_DARK } from '../../../styling/GlobalTheme.style'
const { width, height } = Dimensions.get('screen')

export default StyleSheet.create({
    mainContainer: {
        gap: height * 0.0172,
        padding: width * 0.0446
    },
    inputContainer: {
        gap: height * 0.0092
    },
    text: {
        ...globalThemeStyle.text_Regular,
        color: MAIN_DARK,
        fontSize: 16
    },
})
