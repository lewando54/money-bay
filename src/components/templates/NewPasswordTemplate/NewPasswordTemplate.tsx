import React from 'react'
import { View } from 'react-native'

import { CustomText } from '@atoms'
import { Button } from '@atoms'
import { PasswordInput } from '@molecules'

import newPasswordTemplateStyle from './NewPasswordTemplate.style'

interface INewPasswordTemplateProps {
    newPasswordValue?: string
    confirmPasswordValue?: string
    onNewPasswordChange?: (newValue: string) => void
    onConfirmPasswordChange?: (newValue: string) => void
    onSubmit?: () => void
    testID?: string
}

export default function NewPasswordTemplate({
    newPasswordValue,
    confirmPasswordValue,
    onNewPasswordChange,
    onConfirmPasswordChange,
    onSubmit,
    testID
}: INewPasswordTemplateProps){
    return (
        <View testID={testID} style={newPasswordTemplateStyle.mainContainer}>
            <CustomText style={newPasswordTemplateStyle.text}>Enter new password and confirm.</CustomText>
            <View style={newPasswordTemplateStyle.inputContainer}>
                <PasswordInput value={newPasswordValue} onChange={onNewPasswordChange} placeholder='Enter new password'/>
                <PasswordInput value={confirmPasswordValue} onChange={onConfirmPasswordChange} placeholder='Confirm your password'/>
            </View>
            <Button onClick={onSubmit}>Change password</Button>
        </View>
    )
}
