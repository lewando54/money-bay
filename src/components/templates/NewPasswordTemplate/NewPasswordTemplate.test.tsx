import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import NewPasswordTemplate from './NewPasswordTemplate'
describe('NewPasswordTemplate', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<NewPasswordTemplate />)
    expect(tree).toMatchSnapshot()
  })
})
