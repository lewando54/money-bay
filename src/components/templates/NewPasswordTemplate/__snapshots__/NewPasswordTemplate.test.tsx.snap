// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPasswordTemplate should render correctly 1`] = `
<View
  style={
    {
      "gap": 22.9448,
      "padding": 33.45,
    }
  }
>
  <Text
    style={
      {
        "color": "#1B1D4D",
        "fontFamily": "MulishRegular",
        "fontSize": 16,
      }
    }
  >
    Enter new password and confirm.
  </Text>
  <View
    style={
      {
        "gap": 12.2728,
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": "auto",
            "padding": 10,
            "width": "100%",
          },
          {
            "elevation": 20,
            "shadowColor": "#16196670",
            "shadowOffset": {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.08,
            "shadowRadius": 16,
          },
          [
            {
              "alignItems": "center",
              "borderColor": "white",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 14,
            },
            undefined,
            false,
          ],
        ]
      }
      testID="universal-container"
    >
      <TextInput
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Enter new password"
        placeholderTextColor="#868698"
        secureTextEntry={true}
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#1B1D4D",
              "marginRight": 10,
              "maxWidth": "93%",
              "width": "100%",
            },
            undefined,
            {
              "maxWidth": "100%",
            },
          ]
        }
        testID="passwordtest"
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={10}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "display": "none",
          }
        }
        testID="input-icon"
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": "auto",
            "padding": 10,
            "width": "100%",
          },
          {
            "elevation": 20,
            "shadowColor": "#16196670",
            "shadowOffset": {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.08,
            "shadowRadius": 16,
          },
          [
            {
              "alignItems": "center",
              "borderColor": "white",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 14,
            },
            undefined,
            false,
          ],
        ]
      }
      testID="universal-container"
    >
      <TextInput
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Confirm your password"
        placeholderTextColor="#868698"
        secureTextEntry={true}
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#1B1D4D",
              "marginRight": 10,
              "maxWidth": "93%",
              "width": "100%",
            },
            undefined,
            {
              "maxWidth": "100%",
            },
          ]
        }
        testID="passwordtest"
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={10}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "display": "none",
          }
        }
        testID="input-icon"
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4288142079,
            4279654301,
          ],
          "endPoint": [
            1,
            0.5,
          ],
          "locations": undefined,
          "startPoint": [
            0,
            0.5,
          ],
        }
      }
      style={
        [
          {
            "borderRadius": 10,
            "padding": 15,
          },
          false,
          false,
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "MulishSemiBold",
              "fontSize": 16,
              "textAlign": "center",
            },
            {
              "color": "white",
            },
            false,
            false,
          ]
        }
      >
        Change password
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</View>
`;
